src/main/java/com/rackspace/jenkins_nodepool/JanitorialListener.java

```diff
@@ -56,10 +53,24 @@ public void startJanitor() {
         synchronized (lock) {
             if (janitorThread == null) {
                 janitorThread = new Thread(new Janitor(), "Nodepool Janitor Thread");
-                janitorThread.start();
+
+                if (enabled()) {
+                    janitorThread.start();
+                } else {
+                    LOGGER.log(Level.INFO, "Janitor thread is disabled by configuration and will *not* start");
+                }
             }
         }
     }
+
+    private boolean enabled() {
```

Comment:
Why wouldn't a user want the cleanup thread?

Reply:
A user wouldn't - it's nice for us developers though to examine things before they get clobbered by the cleaner thread.